dspace-api/src/main/java/org/dspace/content/CollectionServiceImpl.java

```diff
@@ -831,4 +836,14 @@ public Collection findByIdOrLegacyId(Context context, String id) throws SQLExcep
     public Collection findByLegacyId(Context context, int id) throws SQLException {
         return collectionDAO.findByLegacyId(context, id, Collection.class);
     }
+
+    @Override
+    public int countTotal(Context context) throws SQLException {
+        return collectionDAO.countRows(context);
+    }
+
+    @Override
+    public List<Map> getCollectionsWithBitstreamSizesTotal(Context context) throws SQLException {
```

Comment:
I think the preferred solution would be to cast to a Map in the DAO and then add an @SupressWarnings("unchecked") annotation. We also make use of this in the AbstractHibernateDAO helper methods, because hibernate doesn't always know what it's getting back (but WE do).
I wouldn't add the method to the AbstractHibernateDAO class though since it isn't a "generic" method and only useful in this use case.

See: https://github.com/DSpace/DSpace/blob/master/dspace-api/src/main/java/org/dspace/core/AbstractHibernateDAO.java#L160 for example.